misc enhancement for 3rd-party storage support feature

Vicente-Cheng opened the pull request:
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
Some minor issues of 3rd-party storage support features need to be fixed

**Solution:**
Improve the 3rd-party storage support features for filesystem storage provisioner

**Related Issue:**
https://github.com/harvester/harvester/issues/7798

**Test plan:**
please use the same test case for 3rd-party storage support features

Vicente-Cheng committed:
charts: update the kubectl version to reduce the duplicated image

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

Vicente-Cheng committed:
controller/pvc: create volumeImportSource to init filesystem PVC

    - Because the dataSourceRef need the same namespace of populator
      (volumeImportSource), we need to create it on-demand.
    - Add mutator for filesystem volume with the annotation
      `harvesterhci.io/volumeForVirtualMachine: "true"`

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

Vicente-Cheng committed:
api/volume: enhance the validation of volume export

    The rules looks below:
    - longhorn v1 volume can export to any storageclass
    - CDI volume cannot export to longhorn v1 storageclass

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

Vicente-Cheng committed:
controller: move the deployment controller to watcher

    - Directly requeue the vmimagedownloader to trigger next
      round

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

Vicente-Cheng committed:
vmimagedownloader: improve the image downloader

    - improve the image download for filesystem storage provisioner

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

WebberHuang1118 reviewed (approved):
LGTM, just a nit, thanks.

Vicente-Cheng committed:
upgrade: applied the CDI CRD before upgrade

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

ihcsim reviewed (approved):
LGTM - just some questions and minor suggestions.

WebberHuang1118 reviewed (approved):
LGTM, thanks.

Vicente-Cheng committed:
controller/storageProfile: add LVM AccessMode/VolumeMode

    - Our lvm provision `lvm.driver.harvesterhci.io` did not match
      the CDI storageprofile table. We need to patch it ourselves.

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

Vicente-Cheng committed:
image/cdi: handle the volumeBindingMode WaitForFirstConsumer

    - REF: https://github.com/kubevirt/containerized-data-importer/blob/main/doc/waitforfirstconsumer-storage-handling.md

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

w13915984028 reviewed (commented):
some questions, thanks.

Vicente-Cheng committed:
pvc/mutator: only path the specific annotation key

    - That will prevent conflict and wrong updates.

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

w13915984028 reviewed (approved):
LGTM, thanks for the confirmation and fix.

The version based crd file management can be improved via https://github.com/harvester/harvester/issues/7736 later.

ibrokethecloud reviewed (approved):
lgtm. thanks for the enhancements.
